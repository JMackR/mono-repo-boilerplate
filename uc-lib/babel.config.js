module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
