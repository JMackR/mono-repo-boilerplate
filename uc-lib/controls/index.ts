export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
