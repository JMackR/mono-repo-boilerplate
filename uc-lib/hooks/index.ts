export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
