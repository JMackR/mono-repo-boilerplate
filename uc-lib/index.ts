export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
