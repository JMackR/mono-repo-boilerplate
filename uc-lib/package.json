{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
