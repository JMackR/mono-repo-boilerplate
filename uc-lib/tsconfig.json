{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
