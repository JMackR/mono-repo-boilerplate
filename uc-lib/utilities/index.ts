export * from "./dimensions-utility"
export * from "./media-query"
