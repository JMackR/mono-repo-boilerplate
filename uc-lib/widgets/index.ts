export * from "./account"
export * from "./attribute-list"
export * from "./breadcrumb"
export * from "./dialog/"
export * from "./empty-state"
export * from "./expanding-text-section"
export * from "./flexible-row"
export * from "./full-screen-spinner"
export * from "./inbox-row"
export * from "./indicator-dots"
export * from "./item-preview"
export * from "./listing-photo-list"
export * from "./pill-list"
export * from "./segmented-control"
export * from "./selectable-row"
export * from "./tooltip"
export * from "./user-profile"
export * from "./search"
