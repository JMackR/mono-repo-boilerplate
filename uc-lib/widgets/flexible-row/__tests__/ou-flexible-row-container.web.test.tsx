import { StyleSheetTestUtils } from "aphrodite/no-important"
import { createSerializer } from "jest-aphrodite-react/no-important"
import { FlexibleRow, FlexibleRowContainer } from "uc-lib"
import React from "react"
import * as renderer from "react-test-renderer"

expect.addSnapshotSerializer(createSerializer(() => StyleSheetTestUtils, { removeVendorPrefixes: true }))

describe("FlexibleRow Snapshot Tests", () => {
  test("Render", () => {
    const tree = renderer.create(
      <FlexibleRowContainer>
        <FlexibleRow mainContent="Text" clickAction={() => {}} />
      </FlexibleRowContainer>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Render with extended margins", () => {
    const tree = renderer.create(
      <FlexibleRowContainer extendMargin={true}>
        <FlexibleRow mainContent="Text" clickAction={() => {}} />
      </FlexibleRowContainer>,
    )
    expect(tree).toMatchSnapshot()
  })
})
