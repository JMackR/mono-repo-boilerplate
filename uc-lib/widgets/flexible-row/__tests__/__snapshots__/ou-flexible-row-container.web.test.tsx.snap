// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexibleRow Snapshot Tests Render 1`] = `
.borderView {
  margin-left: 16px;
  margin-right: 16px;
  background-color: #E2E2E2;
  height: 1px;
  display: flex;
}

.ouflex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: undefined;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.container {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  min-height: 42px;
  height: undefined;
  flex: 1;
  display: flex;
  flex-direction: row;
  user-select: undefined;
  cursor: pointer;
  align-items: center;
  vertical-align: middle;
  background-color: #FFFFFF;
}

.mainContentColumn {
  flex-grow: 1;
  align-self: center;
  flex-direction: column;
}

.rightActionColumn {
  margin-left: 8px;
}

.rightIconColumn {
  display: inline-flex;
  padding-left: 8px;
}

.text-textStyle {
  color: #121212;
  font-size: 16px;
  font-weight: normal;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

<div
  className="ouflex"
>
  <div
    className="borderView"
  />
  <div
    className="container"
    onClick={[Function]}
  >
    <div
      className="mainContentColumn"
    >
      <span>
        <div
          className="text-textStyle"
        >
          Text
        </div>
      </span>
    </div>
    <div
      className="rightActionColumn"
    />
    <div
      className="rightIconColumn"
    >
      <actions-chevron-right-line.svg
        alwaysdark="#121212"
        alwayslight="#FFFFFF"
        basalt="#8A8A8A"
        crystal="#FFFFFF"
        crystalhover="#F7F7F7"
        crystalpressed="#EEEEEE"
        disabled="#FAFAFA"
        glacialblue="#3D98D2"
        granite="#ABABAB"
        larchyellow="#F4BE19"
        larchyellowhover="#F5D624"
        larchyellowpressed="#F3A70C"
        limestone="#E2E2E2"
        limestonehover="#EEEEEE"
        limestonepressed="#D0D0D0"
        obsidian="#121212"
        overlay="rgba(18,18,18,0.3)"
        paintbrushred="#E05666"
        palousegreen="#00A87E"
        palousegreenhighlight="#DEF3EC"
        palousegreenhover="#43B893"
        palousegreenpressed="#00986A"
        quartz="#FAFAFA"
        tintcolor="#ABABAB"
      />
    </div>
  </div>
  <div
    className="borderView"
  />
</div>
`;

exports[`FlexibleRow Snapshot Tests Render with extended margins 1`] = `
.ouflex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: undefined;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.container {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  min-height: 42px;
  height: undefined;
  flex: 1;
  display: flex;
  flex-direction: row;
  user-select: undefined;
  cursor: pointer;
  align-items: center;
  vertical-align: middle;
  background-color: #FFFFFF;
}

.mainContentColumn {
  flex-grow: 1;
  align-self: center;
  flex-direction: column;
}

.rightActionColumn {
  margin-left: 8px;
}

.rightIconColumn {
  display: inline-flex;
  padding-left: 8px;
}

.text-textStyle {
  color: #121212;
  font-size: 16px;
  font-weight: normal;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.borderView {
  margin-left: 0px;
  margin-right: 0px;
  background-color: #E2E2E2;
  height: 1px;
  display: flex;
}

<div
  className="ouflex"
>
  <div
    className="borderView"
  />
  <div
    className="container"
    onClick={[Function]}
  >
    <div
      className="mainContentColumn"
    >
      <span>
        <div
          className="text-textStyle"
        >
          Text
        </div>
      </span>
    </div>
    <div
      className="rightActionColumn"
    />
    <div
      className="rightIconColumn"
    >
      <actions-chevron-right-line.svg
        alwaysdark="#121212"
        alwayslight="#FFFFFF"
        basalt="#8A8A8A"
        crystal="#FFFFFF"
        crystalhover="#F7F7F7"
        crystalpressed="#EEEEEE"
        disabled="#FAFAFA"
        glacialblue="#3D98D2"
        granite="#ABABAB"
        larchyellow="#F4BE19"
        larchyellowhover="#F5D624"
        larchyellowpressed="#F3A70C"
        limestone="#E2E2E2"
        limestonehover="#EEEEEE"
        limestonepressed="#D0D0D0"
        obsidian="#121212"
        overlay="rgba(18,18,18,0.3)"
        paintbrushred="#E05666"
        palousegreen="#00A87E"
        palousegreenhighlight="#DEF3EC"
        palousegreenhover="#43B893"
        palousegreenpressed="#00986A"
        quartz="#FAFAFA"
        tintcolor="#ABABAB"
      />
    </div>
  </div>
  <div
    className="borderView"
  />
</div>
`;
